### DISCUSSION TITLE
P1010: User `snoopforms` was denied access on the database `snoopforms.public`

### DISCUSSION THREAD

**Discussion by bytrangle** [formbricks/formbricks#68] [General Requests] [answered]
I've done everything in the [Getting Started](https://github.com/snoopForms/snoopforms#getting-started) section until `yarn prisma migrate dev`.

Here's the error log I got:

```
Error: P1010: User `snoopforms` was denied access on the database `snoopforms.public`
```
Running `npx prisma db pull` returns the same error.

I know this is not specific to Snoopforms but I don't know much about databases and have spent hours finding an answer. Appreciate any help I can get.

**Comment by mattinannt** [comment, marked as answer]
@bytrangle this looks like a very similar issue to your last one. You set the db user and db password in your .env file but your database can't access the database with this username/password. Maybe there is already another database / postgres instance running in port 5432 with other databases and credentials?

Do you want to develop with snoopForms or just get it up and running on your system? If you just want to get it up and running you can also use the `docker compose` method in the [Deployment Section](https://github.com/snoopForms/snoopforms#deployment). This gives provides you with a database in an isolated docker environment where you don't get conflicts with existing software on your system. To use the docker compose database, you need to set the DATABASE_URL to the one provided in the .env.example:
```DATABASE_URL='postgresql://postgres:postgres@postgres:5432/snoopforms?schema=public'```

**Reply by bytrangle** [reply]
Thank you for your help. I want to check out snoopForms with an interest in making a contribution. I don't know much about databases, but don't mind learning.

**Reply by bytrangle** [reply]
@matthiasnannt Do you think that it is worth mentioning in the docs that one can map port 5432 of the Postgres container to a different port of their host? Because users may want to run a different Postgres instances?

**Reply by mattinannt** [reply]
@bytrangle Great advice. Will add that in the docs :-)
